tests/cupy_tests/linalg_tests/test_solve.py

```diff
@@ -339,3 +340,83 @@ def test_invalid_shape(self):
     def test_invalid_index(self):
         self.check_ind((12, 3, 4), ind=-1)
         self.check_ind((18, 3, 3, 2), ind=0)
+
+
+@testing.parameterize(*testing.product({
+    'shape': [(3, 4, 2, 2), (5, 3, 3), (7, 7)],
+    'nrhs': [None, 1, 8],
+    'lower': [True, False],
+    'order': ['C', 'F'],
+    'overwrite_b': [True, False]
+}))
+class TestChosolve(unittest.TestCase):
+
+    @staticmethod
+    def _solve(a, b):
+        if (
+            numpy.lib.NumpyVersion(numpy.__version__) < "2.0.0"
+            or a.shape[:-1] != b.shape
+        ):
+            return numpy.linalg.solve(a, b)
+        b = b[..., numpy.newaxis]
+        return numpy.linalg.solve(a, b)[..., 0]
+
+    @testing.for_dtypes('fdFD')
+    @testing.numpy_cupy_allclose(atol=1e-5)
```

Comment:
set `contiguous_check` to `False` to avoid comparing strides/contiguity. In general, CuPy does not guarantee following NumPy's contiguity requirement.